models/demos/falcon7b/tt/falcon_causallm.py

```diff
@@ -45,12 +40,14 @@ def __init__(
         self.model_config = model_config
         self.seq_len = seq_len
 
+        self.is_grayskull = is_grayskull()
+
         lm_head_weight = None
         if self.state_dict:
             lm_head_weight = self.state_dict["lm_head.weight"]
             lm_head_weight = torch.transpose(lm_head_weight, -2, -1)
 
-        if self.seq_len > 512:
+        if not self.is_grayskull and self.seq_len > 512:
```

Comment:
Would it be better to keep a OPTIMIZED_MODE_PREFILL setting in model_config and set that to is_grayskull()? If all the is_grayskull calls in the model files are coupled, maybe we should only set it once to avoid confusion in the future.